internal/services/appconfiguration/app_configuration_key_resource.go

```diff
@@ -67,18 +77,29 @@ func (k KeyResource) Arguments() map[string]*pluginsdk.Schema {
 		"value": {
 			Type:     pluginsdk.TypeString,
 			Optional: true,
+			Computed: true,
 		},
 		"locked": {
 			Type:     pluginsdk.TypeBool,
 			Optional: true,
 			Default:  false,
 		},
+		"type": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			Default:  "kv",
+		},
```

Comment:
It looks like this needs a ValidateFunc?

Reply:
Yeah good call. Added.